vector/src/main/res/layout/fragment_thread_list.xml

```diff
@@ -37,6 +37,21 @@
         tools:listitem="@layout/item_thread"
         tools:visibility="gone" />
 
+
+        <ProgressBar
+            android:id="@+id/threadListProgressBar"
+            style="@style/Widget.Vector.ProgressBar.Horizontal"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_vertical"
+            android:indeterminate="true"
+            android:visibility="gone"
+            android:layout_marginTop="-6dp"
```

Comment:
I see why we need this but we have a lint rule for negative margins. Is it a problem to show the tiny progressbar on the first message? If it is we need `tools:ignore="NegativeMargin"` at least.

Reply:
Or `app:layout_constraintBottom_toBottomOf="@id/threadListAppBarLayout"` solves the problem without negative margin?

Reply:
For that known problem I used to wrap it around a fragment, that eliminates the problem, but this adds an extra layer and decrease code readability.  So I prefer to use `-6dp` as an exception in my projects, never had an issue with that.

Regarding `app:layout_constraintBottom_toBottomOf="@id/threadListAppBarLayout"` will not have any difference while the padding is either on top or on the bottom.

P.S. On Jetpack Compose they fixed that issues the horizontal progressBar ! 🥇

Reply:
You could have reused what we are doing in the room list / room timeline: https://github.com/vector-im/element-android/blob/main/vector/src/main/res/layout/view_sync_state.xml#L38

Reply:
Yes I saw it but, nesting affects performance and makes code less readable, I cannot see any benefit from that. Do you have a strong point?